### PR TITLE
lesson-4 homework 20.05 05.09.2021

### PR DESCRIPTION
**Pull request by olennikovandrey**

### CODE REVIEW

File: js/lesson-4.test.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+//'sum' function
+const sum = require('./lesson-4');
+
+test('sum 5 + 2 equals 7', () => {
+    expect(sum(5, 2)).toBe(7);
+});
```
Comment (Dronom) [inline js/lesson-4.test.js:6]: check your test
image

File: js/lesson-4.js [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
 }
-
 console.log(sum(5)(2));
 
 // point 6 of lesson-4 homework
-const colors = ['magenta', 'cyan', 'firebrick', 'springgreen', 'skyblue']; //an array with values ​​is given
-let p = document.getElementsByTagName('p'); //assigning the variable "p" the value of the <p> tag from the HTML document
+const color = ["magenta", "cyan", "firebrick", "springgreen", "skyblue"]; //an array with values ​​is given
+
+let p = document.getElementsByTagName("p"); //assigning the variable "p" the value of the <p> tag from the HTML document
```
Comment (Dronom) [inline js/lesson-4.js:13]: use const

Comment (Dronom) [reply js/lesson-4.js:13]: and update your comment or rename variable: you get not one tag, but all "p"

Comment (olennikovandrey) [reply js/lesson-4.js:13]: 1. const updated;
2. comment updated.

File: js/lesson-2-3.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+test4 = max(3, 4);
+console.log(test4);
+
+//ternary_operator_point_6_2_of_homework_lesson_3
+function max(a, b) {
+    return a > b ? a : b;
+}
+
+console.log('point_7_of_homework_lesson_3');
+let arr = [2, 100, 26, 30, 76, 80, 93, 35, 40, 66] //array declaration
```
Comment (Dronom) [inline js/lesson-2-3.js:103]: use const

File: js/lesson-2-3.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+function min(a, b) { //function min will return the smallest value
+    if (a < b) {
+        return a
+    } else {
+        return b
+    }
+}
+
+test3 = min(3, 4);
+console.log(test3);
```
Comment (Dronom) [inline js/lesson-2-3.js:78]: use let/const for test2, test3

File: js/lesson-2-3.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+console.log('point_6_2_of_homework_lesson_3');
+function max(a, b) { //function min will return the greatest value
+    if (a > b) {
+        return a
+    } else {
+        return b
+    }
+}
+
+test4 = max(3, 4);
```
Comment (Dronom) [inline js/lesson-2-3.js:94]: the same

File: js/lesson-2-3.js (hunk 4 above)
Comment (olennikovandrey) [reply js/lesson-2-3.js:78]: fixed

File: js/lesson-2-3.js (hunk 5 above)
Comment (olennikovandrey) [reply js/lesson-2-3.js:94]: fixed

File: js/lesson-2-3.js (hunk 3 above)
Comment (olennikovandrey) [reply js/lesson-2-3.js:103]: fixed

File: js/lesson-2-3.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        continue; //if the values ​​are strictly equal, then output the value 'true'
+        } else {
+        return false;
+        }
+    }
+    return true;
+}
+
+test = palindrome('blablabla'); //call a function for the check value
+let test2 = palindrome('шалаш');//call a function for the check value
```
Comment (Dronom) [inline js/lesson-2-3.js:63]: do you change test2 anywhere?

File: js/lesson-2-3.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+console.log('point_6_1_of_homework_lesson_3');
+function min(a, b) { //function min will return the smallest value
+    if (a < b) {
+        return a
+    } else {
+        return b
+    }
+}
+
+let test3 = min(3, 4);
```
Comment (Dronom) [inline js/lesson-2-3.js:77]: the same

File: js/lesson-2-3.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+console.log('point_6_2_of_homework_lesson_3');
+function max(a, b) { //function min will return the greatest value
+    if (a > b) {
+        return a
+    } else {
+        return b
+    }
+}
+
+let test4 = max(3, 4);
```
Comment (Dronom) [inline js/lesson-2-3.js:94]: same

**Pull request merged**